# 1- Project Overview

This project focuses on analyzing environmental factors affecting electrical energy output in a Combined Cycle Power Plant.

The objective is to explore operational patterns, identify key factors influencing power generation, and develop data-driven insights using Python, Machine Learning, and Power BI.

# 2- Business Understanding

## Business Objective

The objective of this project is to analyze the relationship between environmental conditions and electrical power output in a Combined Cycle Power Plant.

The analysis aims to identify key factors affecting power generation performance and generate data-driven insights that can support operational optimization and decision-making.

## Business Questions

1. Which environmental factors have the strongest impact on electrical power output?

2. How do changes in temperature, pressure, humidity, and exhaust vacuum affect power generation?

3. What operating conditions are associated with higher or lower power output?

4. Can machine learning models accurately predict electrical power output based on environmental variables?

5. Which factors should be monitored to improve performance understanding?


## Success Criteria

- Perform data quality assessment and prepare the dataset for analysis.
- Identify relationships and patterns affecting power generation.
- Build predictive machine learning models.
- Develop an interactive Power BI dashboard.
- Provide actionable insights based on data analysis.

# 3- Import Laibraries

In [1]:
# Data manipulation
import pandas as pd
import numpy as np

# Display settings
import warnings
warnings.filterwarnings("ignore")

# Status update
print("DONE")

DONE


# 4- Load Dataset

In [2]:
# Load dataset
df = pd.read_excel("/kaggle/input/datasets/osmanmusevih/combined-cycle-power-plant-regression/Combined Cycle Power Plant.xlsx")

# Status update
print("DONE")

DONE


In [3]:
# First 5 raws
df.head()

,AT,V,AP,RH,PE
0,14.96,41.76,1024.07,73.17,463.26
1,25.18,62.96,1020.04,59.08,444.37
2,5.11,39.40,1012.16,92.14,488.56
3,20.86,57.32,1010.24,76.64,446.48
4,10.82,37.50,1009.23,96.62,473.90


In [4]:
# Tuple representing the dimensionality 
df.shape

(9568, 5)

In [5]:
# Column labels
df.columns

Index(['AT', 'V', 'AP', 'RH', 'PE'], dtype='object')

# 5- Inital Exploration

In [6]:
# Concise summary
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   AT      9568 non-null   float64
 1   V       9568 non-null   float64
 2   AP      9568 non-null   float64
 3   RH      9568 non-null   float64
 4   PE      9568 non-null   float64
dtypes: float64(5)
memory usage: 373.9 KB


In [7]:
# Descriptive statistics
df.describe()

,AT,V,AP,RH,PE
count,9568.000000,9568.000000,9568.000000,9568.000000,9568.000000
mean,19.651231,54.305804,1013.259078,73.308978,454.365009
std,7.452473,12.707893,5.938784,14.600269,17.066995
min,1.810000,25.360000,992.890000,25.560000,420.260000
25%,13.510000,41.740000,1009.100000,63.327500,439.750000
50%,20.345000,52.080000,1012.940000,74.975000,451.550000
75%,25.720000,66.540000,1017.260000,84.830000,468.430000
max,37.110000,81.560000,1033.300000,100.160000,495.760000


In [8]:
# Count unique values 
df.nunique()

AT    2773
V      634
AP    2517
RH    4546
PE    4836
dtype: int64

## Intial Observations

The dataset contains 9,568 observations and 5 numerical features representing environmental and operational variables collected from a Combined Cycle Power Plant. All features are stored as float64 data types, and no missing values were detected, indicating that the dataset has good data quality for further analysis.

The statistical summary shows that all variables are within the expected operational ranges provided in the dataset description. The electrical power output ranges between 420.26 MW and 495.76 MW, with an average output of approximately 454.37 MW. Environmental variables, including ambient temperature, pressure, relative humidity, and exhaust vacuum, also show realistic operating ranges.

The number of unique values in each feature indicates that the dataset consists of continuous sensor measurements rather than categorical variables. This confirms that the data represents real-world operational readings and is suitable for statistical analysis, visualization, and machine learning modeling.

# 6- Data Quality Assesment

In [9]:
#  Total number of missing values
df.isnull().sum()

AT    0
V     0
AP    0
RH    0
PE    0
dtype: int64

In [10]:
#  Total number of duplicated values
df.duplicated().sum()

np.int64(41)

## Data Quality Findings

The dataset was assessed for missing values and duplicate records.

No missing values were identified across all features, indicating that the dataset is complete and does not require missing value treatment.

However, 41 duplicate rows were detected. Since duplicate records do not provide additional information and may affect statistical analysis and machine learning model performance, they will be removed during the data cleaning stage.

## Outlier Detection

In [11]:
Q1 = df.quantile(0.25)
Q3 = df.quantile(0.75)

IQR = Q3 - Q1

outliers = ((df < (Q1 - 1.5 * IQR)) | 
            (df > (Q3 + 1.5 * IQR))).sum()

outliers

AT     0
V      0
AP    88
RH    12
PE     0
dtype: int64

## Outlier Detection Findings

The dataset was evaluated for potential outliers using the Interquartile Range (IQR) method.

No outliers were detected in Ambient Temperature (AT), Exhaust Vacuum (V), or Electrical Power Output (PE).

A small number of potential outliers were identified in Ambient Pressure (AP) and Relative Humidity (RH). These observations represent a very small percentage of the dataset and may correspond to valid operating conditions rather than data errors. Therefore, they will be retained for further analysis.

# 7- Data Cleaning

In [12]:
# Tuple representing the dimensionality with duplicated values
df.shape

(9568, 5)

In [13]:
# Remove duplication
df = df.drop_duplicates()

# Tuple representing the dimensionality without duplicated values
df.shape

(9527, 5)

## Data Cleaning Summary

As part of the data cleaning process, duplicate records were removed from the dataset. A total of *41 duplicate rows* were identified and deleted, reducing the dataset size from *9,568* to *9,527* observations.

No missing values were found, and potential outliers were retained because they likely represent valid operational conditions rather than data quality issues. The cleaned dataset is now ready for exploratory data analysis.

In [14]:
# Rename columns for better readability
df.rename(
    columns={
        "AT": "Ambient_Temperature",
        "V": "Exhaust_Vacuum",
        "AP": "Ambient_Pressure",
        "RH": "Relative_Humidity",
        "PE": "Power_Output"
    },
    inplace=True
)

# Verify updated column names
df.columns

Index(['Ambient_Temperature', 'Exhaust_Vacuum', 'Ambient_Pressure',
       'Relative_Humidity', 'Power_Output'],
      dtype='object')

## Column Renaming

The original abbreviated column names were replaced with descriptive names to improve readability and maintain consistency throughout the analysis. This makes the dataset easier to understand and simplifies future analysis and visualization

# 8- Export Clean Dataset

In [15]:
# Save the cleaned dataset for the next stages of the project
df.to_csv("clean_power_plant_data.csv", index=False)

# Status update
print("DONE")

DONE
